packages/nuxt3/src/app/composables/asyncData.ts

```diff
@@ -131,17 +143,17 @@ export function useAsyncData<
     } else if (instance && (nuxt.isHydrating || options.lazy)) {
       // 2. Initial load (server: false): fetch on mounted
       // 3. Navigation (lazy: true): fetch on mounted
-      instance._nuxtOnBeforeMountCbs.push(asyncData.refresh)
+      instance._nuxtOnBeforeMountCbs.push(initialFetch)
     } else {
       // 4. Navigation (lazy: false) - or plugin usage: await fetch
-      asyncData.refresh()
+      initialFetch()
     }
     if (options.watch) {
-      watch(options.watch, () => asyncData.refresh())
+      watch(options.watch, () => initialFetch())
```

Comment:
Currently this prevents `watch` option from triggering updates.

This, for example, doesn't work as expected:

https://github.com/nuxt/framework/blob/1a7b570c82311e04812b1bfc913c455d9218d133/examples/composables/use-async-data/components/CounterExample.vue#L1-L5

was there a reason it shouldn't be treated as a subsequent refresh?

```suggestion
      watch(options.watch, () => asyncData.refresh())
```